src/main/java/cz/medel/roleminimization/graph/boundary/GraphService.java

```diff
@@ -27,25 +27,25 @@
 
     private List<List<Vertex>> mcp = new ArrayList<>();
 
-    public Graph test(Graph bipartiteGraph) {
+    public Graph createEdgeDualGraph(Graph bipartiteGraph) {
         Graph edgeDualGraph = new Graph();
 
         bipartiteGraph.getEdges()
-                .forEach(edgeA -> {
-                    String dualVertexIndex = edgeA.getVertexA().getId() + edgeA.getVertexB().getId();
+                .forEach(edgeX -> {
+                    String dualVertexIndex = edgeX.getVertexA().getId() + edgeX.getVertexB().getId();
                     Vertex dualVertex = edgeDualGraph.addOrGetVertex(dualVertexIndex, VertexType.NONE);
 
                     bipartiteGraph.getEdges()
-                            .forEach(edgeB -> {
-                                if (!edgeA.equals(edgeB)) {
-                                    List<Vertex> leftVertices = List.of(edgeA.getVertexA(), edgeB.getVertexA());
-                                    List<Vertex> rightVertices = List.of(edgeA.getVertexB(), edgeB.getVertexB());
+                            .forEach(edgeY -> {
+                                if (!edgeX.equals(edgeY)) {
+                                    List<Vertex> leftVertices = List.of(edgeX.getVertexA(), edgeY.getVertexA());
+                                    List<Vertex> rightVertices = List.of(edgeX.getVertexB(), edgeY.getVertexB());
```

Comment:
Description	Resource	Path	Location	Type
The method of(Vertex, Vertex) is undefined for the type List	GraphService.java	/role-minimization-develop/src/main/java/cz/medel/roleminimization/graph/boundary	line 41	Java Problem